Remove ExecuTorch copy of Vectorized

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #7042

All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

swolchok committed:
Remove ExecuTorch copy of Vectorized

All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
rebase on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

Commit:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

Commit:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

Commit:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

Commit:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

Commit:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok commented on kernels/optimized/vec/intrinsics.h:
```diff
@@ -1,51 +0,0 @@
-/*
```
hey @dbort does our bc policy require me to undelete all these files? saw you mentioned it recently but I don't know the exact text of the policy

dbort replied:
Policy is at https://pytorch.org/executorch/main/api-life-cycle.html

The goal of the policy is to avoid breaking user code that worked with the previous release. Removing a header could break code that currently works.

Looking at the files that this PR changes, I'm not totally sure if they really qualify as part of our public API; not sure if the headers under kernels/... are meant to be used outside the repo or if they're just for the code around them.

kimishpatel replied:
It could be if they are used to author custom ops out of the repo, but 1) I havent seen any examples in the wild and 2) I dont know if we package this header in our pip package (like pytorch) OR make it public somehow

swolchok replied:
> haven't seen any examples in the wild

[I searched GitHub and found nothing after pruning paths within ExecuTorch and forks.](https://github.com/search?q=executorch%2Fkernels%2Foptimized%2Fvec+NOT+repo%3Apytorch%2Fexecutorch++NOT+path%3Aexecutorch+NOT+repo%3Ajustinchuby%2Fexecutorch+NOT+repo%3Akirklandsign%2Fexecutorch+NOT+repo%3ADenisVieriu97%2Fexecutorch_fork+NOT+repo%3AGregoryComer%2Fexecutorch+NOT+repo%3ACodeLinaro%2Fexecutorch+NOT+repo%3Acopyrightly%2Fexecutorch+NOT+path%3Akernels%2Foptimized%2Fcpu+NOT+path%3Aextension%2Fllm%2Fcustom_ops+NOT+path%3Akernels%2Foptimized%2Fvec+NOT+path%3Aexamples%2Fmodels+NOT+path%3Akernels%2Foptimized%2Ftest&type=code)

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok commented:
This will require https://github.com/pytorch/pytorch/pull/152366 to be included in a pin bump before landing

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
Update on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
fix numerics for internal test on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok commented:
llava-runner-linux result is slightly different but wouldn't call it functionally incorrect. rerunning failed jobs but we may need to discuss whether this is landable. https://github.com/pytorch/executorch/actions/runs/15572676327/job/43851619080?pr=7042

swolchok commented:
yep, need to update test-llava-runner-linux output https://github.com/pytorch/executorch/actions/runs/15572676327/job/43855070881?pr=7042

swolchok committed:
shorten tested llava prefix due to output change due to slightly changed numerics on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]

swolchok committed:
band-aid unittest-buck on "Remove ExecuTorch copy of Vectorized"


All uses are outside ExecuTorch core, so we can just use ATen Vectorized.

Differential Revision: [D66396016](https://our.internmc.facebook.com/intern/diff/D66396016/)

[ghstack-poisoned]
